plugins/woocommerce/src/StoreApi/Schemas/V1/CartItemSchema.php

```diff
@@ -79,6 +79,30 @@ public function get_item_response( $cart_item ) {
 		];
 	}
 
+	/**
+	 * Get list of product images for the cart item.
+	 *
+	 * @param \WC_Product $product       Product instance.
+	 * @param array       $cart_item     Cart item array.
+	 * @param string      $cart_item_key Cart item key.
+	 * @return array
+	 */
+	protected function get_cart_images( \WC_Product $product, array $cart_item, string $cart_item_key ) {
+		// Get the images for the product
+		$product_images = $this->get_images( $product );
+
+		/**
+		 * Filter the cart product images
+		 *
+		 * This hook allows for changing the cart item images. This is specific to the cart endpoint.
+		 *
+		 * @param array  product_images  Array of image objects, as defined in ImageAttachmentSchema
+		 * @param array  $cart_item      Cart item array.
+		 * @param string $cart_item_key  Cart item key.
+		 */
+		return apply_filters( 'woocommerce_cart_item_images', $product_images, $cart_item, $cart_item_key );
+	}
```

Comment:
I had another chat with @opr, and we believe that we should save guard the filter. If the filter does not return a valid image, the original image should be used as a fell back. Such a safeguard could look like this:

```php
	protected function get_cart_images( \WC_Product $product, array $cart_item, string $cart_item_key ) {
		// Get the images for the product
		$product_images = $this->get_images( $product );

		/**
		 * Filter the cart product images
		 *
		 * This hook allows for changing the cart item images. This is specific to the cart endpoint.
		 *
		 * @param array  product_images  Array of image objects, as defined in ImageAttachmentSchema
		 * @param array  $cart_item      Cart item array.
		 * @param string $cart_item_key  Cart item key.
		 */
		$filtered_images = apply_filters( 'woocommerce_cart_item_images', $product_images, $cart_item, $cart_item_key );

		// Return the original images if the filtered image has no thumbnail URL.
		foreach ( $filtered_images as $key => $image ) {
			if ( empty( $image->thumbnail ) || ! filter_var( $image->thumbnail, FILTER_VALIDATE_URL ) ) {
				return $product_images;
			}
		}

		// Return the filtered images.
		return $filtered_images;
	}
```

Reply:
Would you want the guard to first filter out images without thumbnail/src, and return the remaining guarded filtered image array, or on an single filtered image failing the guard, return the original array? Cause the provided guard sample will fall back to the original array if a single image fails the guard. Might be intended, just wanted to make sure.

For example:
``` PHP
protected function get_cart_images( \WC_Product $product, array $cart_item, string $cart_item_key ) {
	// Get the images for the product
	$product_images = $this->get_images( $product );

	/**
	 * Filter the cart product images
	 *
	 * This hook allows for changing the cart item images. This is specific to the cart endpoint.
	 *
	 * @param array  product_images  Array of image objects, as defined in ImageAttachmentSchema
	 * @param array  $cart_item      Cart item array.
	 * @param string $cart_item_key  Cart item key.
	 */
	$filtered_images = apply_filters( 'woocommerce_cart_item_images', $product_images, $cart_item, $cart_item_key );

	// Return the original images if the filtered image has no thumbnail URL.
	$guarded_images = array();
	foreach ( $filtered_images as $key => $image ) {
		// Check if thumbnail is a valid url
		if ( empty( $image->thumbnail ) || ! filter_var( $image->thumbnail, FILTER_VALIDATE_URL ) ) {
			continue;
		}
		// Check if src property is a valid url
		if ( empty( $image->src) || ! filter_var( $image->src, FILTER_VALIDATE_URL ) ) {
			continue;
		}

		// Image is valid, add to resulting array
		$guarded_images[] = $image;
	}

	// If there are no valid images remaining, return original array
	if ( count($guarded_images) == 0 ) {
		return $product_images;
	}

	// Return the filtered guarded images.
	return $guarded_images;
}
```

Reply:
Also, when failing a guard check, maybe add a log for it? So plugin developers can see in logs why it is failing?

``` PHP
$logger = wc_get_logger();
foreach ( $filtered_images as $key => $image ) {
	// Check if thumbnail is a valid url
	if ( empty( $image->thumbnail ) || ! filter_var( $image->thumbnail, FILTER_VALIDATE_URL ) ) {
		$logger->warning( "After passing through woocommerce_cart_item_images filter, image with id $image->id 
 did not have a valid thumbnail property." );
		continue;
	}
	// Check if src property is a valid url
	if ( empty( $image->src) || ! filter_var( $image->src, FILTER_VALIDATE_URL ) ) {
		$logger->warning( "After passing through woocommerce_cart_item_images filter, image with id $image->id 
 did not have a valid src property." );
		continue;
	}

	// Image is valid, add to resulting array
	$guarded_images[] = $image;
}
```

Reply:
> Would you want the guard to first filter out images without thumbnail/src, and return the remaining guarded filtered image array, or on an single filtered image failing the guard, return the original array? Cause the provided guard sample will fall back to the original array if a single image fails the guard. Might be intended, just wanted to make sure.

Great catch, @synercoder. We should display images from the filter as long as they're valid. Logging invalid images, also makes sense, IMO. Could you implement these changes, and apply the coding styles?

Reply:
@nielslange @opr I added the proposed changes to the branch.